### PR TITLE
Treat Build Warnings as Errors for Abstractions & DependencyInjection Projects

### LINKED ISSUE #4040: Treat Warnings as Build Errors - DotNetNuke.Abstractions
## Description of problem
DNN has many build warnings riddled throughout the entire solution which are usually indicators of issues with the code. Be it legacy code that has not been cleaned up or wasted memory footprints when allocating data that is never used. We should be treating all build warnings as errors to ensure a high level of code quality.

This is a massive undertaking with the size of DNN, I recommend we do this for the `DotNetNuke.Abstractions` project as it is the newest project and smallest project.

## Description of solution
- Update `.csproj` to treat build warnings as erros
- Update code to remove all build errors

## Description of alternatives considered
N/A

## Screenshots
N/A

## Additional context
Code Quality Enhancement

## Affected browser
N/A

### PR DESCRIPTION
**Pull request by ahoefling**
Fixes #4040 

## Summary
This PR updates `DotNetNuke.Abstractions` and `DotNetNuke.DependencyInjection` to treat all build warnings as errors. This will help ensure a high level of code quality in these new projects.

### CODE REVIEW

File: DNN Platform/DotNetNuke.Abstractions/Prompt/ICommand.cs [hunk 1]
```diff
@@ -1,37 +1,45 @@
...
         /// <summary>
-        /// Description of the command
+        /// Gets or sets the description of the command.
         /// </summary>
         string Description { get; set; }
+
         /// <summary>
-        /// Category to which this command belongs. This is used to 
-        /// group the list of commands when a user requests this.
+        /// Gets or sets thecategory to which this command belongs.
```
Comment (bdukes) [inline DNN Platform/DotNetNuke.Abstractions/Prompt/ICommand.cs:23]:
```suggestion
        /// Gets or sets the category to which this command belongs.
```

File: DNN Platform/DotNetNuke.Abstractions/Prompt/ICommandHelp.cs [hunk 2]
```diff
@@ -1,33 +1,38 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information
+
```
Comment (bdukes) [inline DNN Platform/DotNetNuke.Abstractions/Prompt/ICommandHelp.cs:4]: Is this blank line required?  I had been removing this line.  I'm fine either way, just wondering if I was working against StyleCop when I was doing that.

**Review by bdukes — APPROVED** [review]
💪🏻

File: DNN Platform/DotNetNuke.Abstractions/Prompt/ICommandHelp.cs (hunk 2 above)
Comment (ahoefling) [reply DNN Platform/DotNetNuke.Abstractions/Prompt/ICommandHelp.cs:4]: It is not a stylecop setting, for some reason I thought it was inside our stylecop configuration. I can remove these blank lines below the file header.

Comment (bdukes) [reply DNN Platform/DotNetNuke.Abstractions/Prompt/ICommandHelp.cs:4]: I would prefer no white space, but happy to be overridden if there are any other opinions

Comment (ahoefling) [reply DNN Platform/DotNetNuke.Abstractions/Prompt/ICommandHelp.cs]: I have removed the whitespace, I'm going to leave this conversation open if anyone has strong opinions on this

**Comment by valadas** [comment]
image

**Comment by david-poindexter** [comment]
OMG @valadas - who does that???????  🤣

**Review by valadas — APPROVED** [review]
Looks good to me

**Pull request merged**